check_http_json.rb

```diff
@@ -426,115 +460,127 @@ def sanity_check(options)
         end
     end
     # Build a nice output string (issue #17).
-    perf = ' | ' + p.join(' ')
+    Nagios.perf = ' | ' + p.join(' ')
 end
 
+# ensure element is an array
+options[:element] = []
+
+# used in ok message to represent configured check element
+element_message_name = ''
+
 # If the element is a string...
-if options[:element_string] then
-    if not json_flat.has_key?(options[:element_string]) then
-        # Not sure if this should be WARN or CRIT. --phrawzty
-        msg = '%s not found in response.' % [options[:element_string]] + perf
-        do_exit(options[:v], 1, msg)
-    end
+unless options[:element_string].empty?
+    element_message_name = options[:element_string].join(',')
     options[:element] = options[:element_string]
 end
 
 # If the element is a regex...
-if options[:element_regex] then
-    json_flat.each do |k,v|
-        if k =~ Regexp.new(options[:element_regex]) then
-            say(options[:v], "Found %s as %s" % [options[:element_regex], k])
-            options[:element] = k
-        end
+if options[:element_regex]
+    element_message_name = 'First'
+    element_message_name = 'All' if options[:element_regex_global]
+    element_message_name = "%s '%s' (regex)" % [element_message_name, options[:element_regex]]
+
+    json_flat.each do |k, _|
+        next unless k =~ Regexp.new(options[:element_regex])
+
+        say(options[:v], 'Found %s as %s' % [options[:element_regex], k])
+        options[:element].push k
+        # do not add all elements if not enabled
+        break unless options[:element_regex_global]
     end
-    if not options[:element] then
-        msg = '%s not found in response.' % [options[:element_regex]] + perf
-        do_exit(options[:v], 3, msg)
+    if options[:element].empty?
+        msg = '%s not found in response.' % [options[:element_regex]]
+        Nagios.critical = msg
     end
 end
 
-say(options[:v], 'The value of %s is %s' % [options[:element], json_flat[options[:element]]])
-
-# If we're looking for a string...
-if options[:result_string] then
-    if json_flat[options[:element]].to_s == options[:result_string].to_s then
-        msg = '%s is %s' % [options[:element], json_flat[options[:element]]] + perf
-        do_exit(options[:v], 0, msg)
-    else
-        msg = '%s is %s' % [options[:element], json_flat[options[:element]]] + perf
-        do_exit(options[:v], 2, msg)
-    end
+# build ok message
+if options[:result_string]
+    Nagios.ok = '%s does match %s', [element_message_name, options[:result_string]]
```

Comment:
You probably meant to use double-quotes here; with the singles, the string is literal, which gives:
```
(pr/33) $ ./check_http_json.rb -u http://jsonplaceholder.typicode.com/users -e 1.id -r 2
OK: %s does match %s1.id2
```